addons/stock_picking_batch/models/stock_move.py

```diff
@@ -12,3 +12,13 @@ def _search_picking_for_assignation_domain(self):
         domain = super()._search_picking_for_assignation_domain()
         domain = expression.AND([domain, ['|', ('batch_id', '=', False), ('batch_id.is_wave', '=', False)]])
         return domain
+
+    def _action_cancel(self):
+        res = super()._action_cancel()
+        cancelled_move_pickings = self.filtered(lambda m: m.state == 'cancel').mapped('picking_id')
+
+        for picking in cancelled_move_pickings:
+            # The picking is cancelled if every moves within it is cancelled as well.
+            if picking.state == 'cancel' and picking.batch_id and any(picking.state != 'cancel' for picking in picking.batch_id.picking_ids):
+                picking.batch_id = None
+        return res
```

Comment:
To make it more readable, it's better to avoid same variable name for different variables in nested loops (`picking` in your case)

Reply:
Indeed, seems I forgot this one. Thanks for pointing that out !